activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -53,16 +53,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
     Relation::MULTI_VALUE_METHODS.each do |name|
       class_eval <<-CODE, __FILE__, __LINE__ + 1
-        def #{name}_values                    # def select_values
-          @values[:#{name}] || []             #   @values[:select] || []
-        end                                   # end
-                                              #
-        def #{name}_values=(values)           # def select_values=(values)
-          assert_mutability!                  #   assert_mutability!
-          @values[:#{name}] = values          #   @values[:select] = values
-        end                                   # end
+        def #{name}_values                        # def select_values
+          @values[:#{name}] || FROZEN_EMPTY_ARRAY #   @values[:select] || []
+        end                                       # end
+                                                  #
+        def #{name}_values=(values)               # def select_values=(values)
+          assert_mutability!                      #   assert_mutability!
+          @values[:#{name}] = values              #   @values[:select] = values
+        end                                       # end
```

Comment:
We've been dropping this commentary style as we make changes that touch it. Feel free to axe here too.